### ISSUE TITLE
Cstor volume target deployment stuck in Terminating state

### ISSUE THREAD

**Issue by princerachit** [openebs/openebs#1886]
## Is this a BUG REPORT or FEATURE REQUEST?
 
 BUG REPORT

**What happened**:
- Deleted a PVC of openebs. Volume type is CStor.
- The target deployment is stuck in terminating phase.
- All other resources corresponding to that volume.

**What you expected to happen**:
- Target deployment gets deleted successfully

**How to reproduce it (as minimally and precisely as possible)**:
- Create a cstor volume.
- Delete the pvc corresponding to that volume.

**Environment**:
```
prince@prince:~/go/src/github.com/princerachit/dump/vol-test/1$ kubectl get deployments.apps,po -n openebs 
NAME                                                    DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/cstor-pool-default-0.7.0-7vqx           1         1         1            1           2h
deployment.apps/cstor-pool-default-0.7.0-w9ie           1         1         1            1           2h
deployment.apps/cstor-pool-default-0.7.0-y9vh           1         1         1            1           2h
deployment.apps/default-percona-pvc-1773207724-target   1         0         0            0           2h
deployment.apps/maya-apiserver                          1         1         1            1           2h
deployment.apps/openebs-provisioner                     1         1         1            1           2h
deployment.apps/openebs-snapshot-operator               1         1         1            1           2h

NAME                                                       READY     STATUS        RESTARTS   AGE
pod/cstor-pool-default-0.7.0-7vqx-bc657df8d-rphfp          2/2       Running       0          2h
pod/cstor-pool-default-0.7.0-w9ie-7d8bdf47d7-kjpmx         2/2       Running       0          2h
pod/cstor-pool-default-0.7.0-y9vh-58dd8df9b9-jmhxg         2/2       Running       0          2h
pod/default-percona-pvc-1773207724-target-f68c546d-ltpcf   0/3       Terminating   0          2h
pod/maya-apiserver-7587997f4c-8v4ss                        1/1       Running       0          2h
pod/openebs-ndm-hndww                                      1/1       Running       0          2h
pod/openebs-ndm-qhfzb                                      1/1       Running       0          2h
pod/openebs-ndm-qtrrw                                      1/1       Running       0          2h
pod/openebs-provisioner-6fd4c8cfd-gds84                    1/1       Running       0          2h
pod/openebs-snapshot-operator-7f7499d96b-jqbxk             2/2       Running       0          2h

```
- OS (e.g. from /etc/os-release): `Ubuntu 16.04.5`
- Kernel : `Linux gke-prince-cluster-pool-1-1ddefaed-lpwf 4.15.0-1017-gcp #18~16.04.1-Ubuntu SMP Fri Aug 10 13:26:07 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`
- Openebs release- `0.7.0`
- Kubernetes version: Running GKE version `1.10.6-gke.2`

**Possible workaround**
Deleting the deploy and pod forcefully will clean up the resource.
1. First delete the target deploy:
`kubectl delete deploy <volume-target-deploy> -n openebs --force --grace-period=0`
2. Then delete the pod:
`kubectl delete po <volume-target-po> -n openebs --force --grace-period=0`

**Comment by princerachit** [comment]
@kmova This issue has been resolved with https://github.com/openebs/maya/pull/512 PR. Shall we close this issue?

**Comment by kmova** [comment]
Merged into 0.7.1 branch.
